### ISSUE TITLE
ImagePicker orientation iOS (portrait only)

### ISSUE THREAD

**Issue by dithom** [expo/expo#2020]
I'm experiencing some strange behavior with the ImagePicker only appearing in portrait orientation. My app is landscape only as set in app.json.

    {
        "expo": {
            "sdkVersion": "25.0.0",
            "orientation": "landscape",
            "platforms": [
                "ios"
            ],
            "privacy": "unlisted",
            "splash": {
                "image": "./assets/splashscreen.png"
            },
            "icon": "./assets/app_icon.png",
            "slug": "my-app",
            "name": "my-app"
        }
    }

I'm having this issue on SDK 25 on iOS iPad Air 2. When I started the project everything worked fine, but it seems since the "no-more-barcode-on-iOS"-update of the Expo client rolled out, this issue started to appear. I'm trying to open the ImagePicker modal in landscape, but its opening in portrait mode. The app itself stays landscape though. The ImagePicker modal even moves back down to bottom correctly.

    let result = await ImagePicker.launchImageLibraryAsync({
        mediaTypes: "Images",
        base64: true,
        quality: 1
    });

Heres a link to a screencast: https://youtu.be/LA_jKWh2tC4

Another thing I was wondering about is if it is possible to change the text of the "Cancel" label in the header of the ImagePicker.

**Comment by ccheever** [comment]
Thanks for the report!

In the meantime while this gets looked at, your best work around is probably to use the Camera module which will give you control over everything so you can make it the way you want it.

**Comment by dithom** [comment]
@ccheever Is there any workaround to get the photo roll displayed correctly at this time?

**Comment by luketanner** [comment]
Likewise — this has become a breaking issue for us. Has anyone figured out a workaround?

**Comment by jlorencelim** [comment]
Experiencing this issue as well. Are there any updates on this?

**Comment by jlorencelim** [comment]
Experienced this issue as well but was able to fix this issue by changing the `UIImagePickerController`'s `modalPresentationStyle` from `UIModalPresentationOverFullScreen`  to `UIModalPresentationOverCurrentContext`.

```
self.picker.modalPresentationStyle = UIModalPresentationOverCurrentContext;
```

**Comment by dithom** [comment]
@jlorencelim Could you please provide a detailed explanation, how to fix this issue with expo? That would be awesome!

**Comment by jlorencelim** [comment]
@dithom I just changed the code in `EXImagePicker.m` from 
```
self.picker.modalPresentationStyle = UIModalPresentationOverFullScreen;
```
to 
```
self.picker.modalPresentationStyle = UIModalPresentationOverCurrentContext;
```

It worked fine but I am not sure if there would be problems be showing with this fix. So far, it just worked fine.

**Comment by dithom** [comment]
@jlorencelim Thanks for the fast response! So the reason I didn't get it the first time was that I'm apparently missing `EXImagePicker.m`... Do I have to eject to ExpoKit first? (As the ExpoSDK is provided by the Expo Client App and therefore can't be modified?!?) Or am I missing something? The only `.m` I can find is `RCTImagePickerManager.m` but any changes in this file have no effect on neither `ImagePicker`provided by Expo nor `ImagePickerIOS` provided by React-Native.

I hope there's a workaround without ejecting, as I ran into a lot of problems trying to eject to ExpoKit...

**Comment by jlorencelim** [comment]
@dithom unfortunately I had my app ejected using ExpoKit. I had to manually change the code in the Pods to make it work.

**Comment by dithom** [comment]
@jlorencelim hm ok so either eject and change it in the Pods or wait for Expo to fix it in their modules...

**Comment by lmichailian** [comment]
Some news about this ?

**Comment by dithom** [comment]
Doesn't seem to be fixed with this commit #3174
